x/pubsub/keeper/proposal_handler.go

```diff
@@ -0,0 +1,43 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/peggyjv/sommelier/v3/x/pubsub/types"
+)
+
+// HandleAddPublisherProposal is a handler for executing a passed community publisher addition proposal
+func HandleAddPublisherProposal(ctx sdk.Context, k Keeper, p types.AddPublisherProposal) error {
+	// TODO(bolten): I assume that for a proposal to reach this point it has already have ValidateBasic called on it?
+	_, found := k.GetPublisher(ctx, p.Domain)
+	if found {
+		// TODO(bolten): should we just overwrite an existing publisher if governance passes like this, or should we
+		// require that the existing publisher is removed first?
```

Comment:
At proposal creation time couldn't we

1. Check if the publisher is already registered with this domain
2. Check if there are any other active proposals for this publisher with this domain
3. Reject the proposal if either of the above return the affirmative

and then not have to check here?

Reply:
After speaking with Zaki it doesn't seem like the governance router currently has the capability of doing this.